Change timeouts to be configurable

ygaller opened the pull request:
I would like to externalize the configuration of timeouts. This allows the sender to "fail fast" in case the Zipkin server is hung, which can be useful during startup, in case you don't want to wait the default readTimeout of 60 seconds for the server to come up.

adriancole commented:
looks alright though the build is broke. Maybe fix the accidental formatting?

Also, is it possible to use null to avoid copy/pasting default values similar to here? https://github.com/openzipkin/zipkin-reporter-java/blob/master/spring-beans/src/main/java/zipkin2/reporter/beans/URLConnectionSenderFactoryBean.java

finally, not sure if we have any yet, but unit tests are also lovely. ex but obviously these would be different https://github.com/openzipkin/zipkin-reporter-java/blob/master/spring-beans/src/test/java/zipkin2/reporter/beans/URLConnectionSenderFactoryBeanTest.java#L60

ygaller commented:
@adriancole  Changes:
- Fixed the accidental formatting
- Changed values to be nullable
- Updated the hello-world application and readme.md to include the timeout fields

Regarding unit tests / broken build: 

Unit tests today are minimal (test that the http factory is discoverable). They do not include any mocking of the factory, so adding proper test coverage and mocks is a mini-project in itself. 

The build is broken due to the upgrade to JDK 10 in the previous PR. The JDK 8 build passes, so I think fixing the JDK 10 issue is a separate issue.

ygaller commented:
Seems that JDK 10 is not supported yet: https://github.com/travis-ci/travis-ci/issues/9368

adriancole commented:
Thanks for all the help!

jplock commented on zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/HttpZipkinFactory.java:
```diff
@@ -51,6 +56,22 @@ public void setBaseUrl(String baseUrl) {
     }
 
+    @Nullable
+    private Integer connectTimeout;
```
Sorry, just seeing this PR now. Can we use Dropwizard’s Duration class in both of these so the use can control the TimeUnit. Otherwise, looks good.

ygaller replied:
@jplock Wasn't aware of this option. Nice.

jplock commented:
Travis doesn’t support jdk10 which is what I think is failing the build. Change looks good.
